File: 프로그래머스/0/181854. 배열의 길이에 따라 다른 연산하기/배열의 길이에 따라 다른 연산하기.js
function solution(arr, n) {
    let answer = Array.from(arr)

    for (let i = 0; i < answer.length; i++) {
        if (answer.length % 2 == 0) {
            i % 2 == 1 && (answer[i] += n)
        } else {
            i % 2 == 0 && (answer[i] += n)
        }
    }

    return answer
}

File: 프로그래머스/0/181854. 배열의 길이에 따라 다른 연산하기/배열의 길이에 따라 다른 연산하기.java
class Solution {
    public int[] solution(int[] arr, int n) {
        int[] answer = new int[arr.length];
        
        for (int i = 0; i < arr.length; i++) {
            int val = arr[i];
            if (arr.length % 2 == 0) {
                answer[i] = i % 2 == 1 ? val + n : val;
            } else {
                answer[i] = val;
                answer[i] += i % 2 == 0 ? n : 0;
            }
        }
        
        return answer;
    }
}